d3-series/materials/Week 4/Assignment/js/climate_chart.js

```diff
@@ -0,0 +1,166 @@
+    var svg = d3
+        .select(containerId)
+        .append('svg')
+        .attr('height', height)
+        .attr('width', width);
+
+    // create inner group element
+    var g = svg
+        .append('g')
+        .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+
+    
+        // scales
+        
+        var x = d3
+            .scaleBand()
+            .domain(
+                data.map(function(d) {
+                    return d.year;
+                })
+            )
+            .range([0, innerWidth])
+            .padding(0.4);
+
+        console.log(x.domain(), x.range());
+
+        var y = d3
+            .scaleLinear()
+            .domain([
+                d3.min(data, function(d){
+                    return d.temp*1.05;
+                }),
+                d3.max(data, function(d) {
+                    return d.temp*1.05;
+                })
+            ])
+            .range([innerHeight, 0]);
+
+        //console.log(y.domain(), y.range());
+
+        // axes
+        var decades = x.domain().filter(function(d, i) { return !(i % 10); })
+        var xAxis = d3.axisBottom(x)
+            .tickValues(decades)
+
+        g
+            .append('g')
+            .attr('class', 'x-axis')
+            .attr('transform', 'translate(0,' + innerHeight + ')')
+            .call(xAxis);
+
+        var yAxis = d3.axisLeft(y).ticks(10);
+
+        g
+            .append('g')
+            .attr('class', 'y-axis')
+            .call(yAxis);
+
+        // bars
+        var colors = d3.scaleLinear()
+            .domain(decades)
+            .range(["01015c", "#00008b","#17008b", "#2e008b",
+                    "#5d008b", "#74008b", "#8b008b", "#8b0074",
+                    "#8b005d", "#8b0045" ,"#8b002e", "#8b0017", 
+                    "#8b0000", "#6e0301"]);
+        console.log(colors)
+
+        g
+            .selectAll('.bar')
+            .data(data)
+            .enter()
+            .append('rect')
+            .attr('class', 'bar')
+            .attr("fill", data=>colors(data.year))
+            /*.attr("fill", function(d) {
+                if (d.temp <= 0) {
+                  return "red";
+                } else  {
+                  return "steelblue";
+                }
+              })*/
+            .attr('x', function(d) {
+                return x(d.year);
+            })
+            .attr('y', (d) => y(Math.max(0, d.temp)))
+            .attr('width', x.bandwidth())
+            .attr('height', d => Math.abs(y(d.temp) - y(0)))
+            .attr('stroke', 'none');
+
+
+        var createHorizontalLine = function(y_pos, line_length){
+            g
+                .append("g")
+                .attr("transform", "translate(0, "+ y(y_pos) +")")
+                .append("line")
+                .attr("x2", line_length)
+                .style("stroke", "black")
+                .style("stroke-width", "1px")
+        };
+
+        createHorizontalLine(0, innerWidth);
```

Comment:
same as for the c02 emissions, you are passing the wrong parameter for the ordinate here